src/mscorlib/src/System/Reflection/Emit/ILGenerator.cs

```diff
@@ -1174,6 +1174,25 @@ public virtual void ThrowException(Type excType)
             this.Emit(OpCodes.Throw);
         }
 
+        private static Type GetConsoleType()
+        {
+#if PLATFORM_UNIX
+            Type t = null;
+            try
+            {
+                t = Type.GetType("System.Console, System.Console, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", throwOnError: false);
+            }
+            catch { } // even with throwOnError:false, could still throw for a variety of reasons
```

Comment:
The documentation is accurate. I think these exceptions should propagate because they are in cases where the assembly is found but can't be used, which isn't expected on any platform.

TargetInvocationException or BadImageFormatException would mean that System.Console.dll is plain broken somehow, which I'd want to debug.

FileLoadException is debatable because it can throw if the version found was too low, and that is arguably the same as not finding it on the given platform by the contract versioning scheme we have. We have shipped 3.9 "pre-versions" before, but it's exotic enough that I'm still not sure that we should replace the detail of this mismatch with an empty PlatformNotSupportedException.

One related thing I hadn't considered in the previous review is that failure to find System.Console.dll isn't necessarily a platform issue in a world of app-local deployment. Maybe throwOnError: true without catch would be most likely to help flag that System.Console.dll must be deployed for this API to work?